From stable-bounces@linux.kernel.org Tue Dec  6 23:50:21 2005
Date: Wed, 7 Dec 2005 08:46:33 +0100
From: Dominik Brodowski <linux@dominikbrodowski.net>
To: stable@kernel.org
Message-ID: <20051207074633.GA26768@dominikbrodowski.de>
Content-Disposition: inline
Cc: 
Subject: i82365: release all resources if no devices are found

From: Igor Popik <igor.popik@gmail.com>

The i82365 driver does not release all the resources when the device is not
found. This can cause an oops when reading /proc/ioports after module
unload (e.g. bug #5657).

Signed-off-by: Igor Popik <igor.popik@gmail.com>
Signed-off-by: Dominik Brodowski <linux@dominikbrodowski.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/pcmcia/i82365.c |    1 +
 1 file changed, 1 insertion(+)

--- linux-2.6.14.3.orig/drivers/pcmcia/i82365.c
+++ linux-2.6.14.3/drivers/pcmcia/i82365.c
@@ -1382,6 +1382,7 @@ static int __init init_i82365(void)
     if (sockets == 0) {
 	printk("not found.\n");
 	platform_device_unregister(&i82365_device);
+	release_region(i365_base, 2);
 	driver_unregister(&i82365_driver);
 	return -ENODEV;
     }
